packages/manager/apps/web/client/app/components/hosting-domain-offers/controller.js

```diff
@@ -99,20 +112,74 @@ export default class WebComponentsHostingDomainOffersController {
     const groupedOffers = groupBy(offers, 'category');
     this.groupedOffers = Object.keys(groupedOffers).map((category) => {
       const versions = groupedOffers[category];
+      const equalOffer = versions.find(
+        ({ badge }) => badge.type === BADGES.EQUAL,
+      );
+      const selectedVersion = equalOffer || versions[0];
 
       return {
         category,
         versions,
-        selectedVersion: versions[0],
-        price: versions[0].price,
+        selectedVersion,
+        price: selectedVersion.price,
       };
     });
 
+    // preselect equal offer (if exist)
     this.model = {
-      offer: null,
+      offer: this.getEqualOffer(),
     };
   }
 
+  static buildBadgeModel(type, className) {
+    return {
+      type,
+      className,
+    };
+  }
+
+  static getOfferPrice(offer) {
+    return offer.versions.length > 1
+      ? offer.selectedVersion.price
+      : offer.price;
+  }
+
+  getEqualOffer() {
+    return this.groupedOffers.find(
+      ({ selectedVersion }) => selectedVersion.badge.type === BADGES.EQUAL,
+    );
+  }
+
+  getOfferBadge(offer) {
+    const Ctrl = WebComponentsHostingDomainOffersController;
```

Comment:
function buildBadgeModel and getOfferPrices should be move into an (helper) service to avoid doing this

Reply:
We could use `WebComponentsHostingDomainOffersController` class name but i prefer to have a shorthand version.
May be you have something in mind ?
Don't hesitate to let suggestions!

Reply:
Static functions are the way to go IMO
You could use `this.constructor`

Reply:
Doing `Ctrl.theStaticMethod()` equal 
`WebComponentsHostingDomainOffersController.theStaticMethod()` equal
`this.constructor.theStaticMethod()`

I choose first one because shortest to use.

Reply:
Usually, I prefer moving util functions to service in order to keep controller lighter. But if your functions are only needed here, no problem, it's ok for me

Reply:
not required a service on this case.